test_util/test_upgrade_vpc.py

```diff
@@ -235,6 +247,13 @@ def setup_cluster_workload(self, dcos_api: DcosApiSession, healthcheck_app: dict
             for app in test_apps:
                 assert app['instances'] == len(self.tasks_start[app['id']])
 
+            # Save the master's state of the task to compare with
+            # the master's view after the upgrade.
+            # See this issue for why we check for a difference:
+            # https://issues.apache.org/jira/browse/MESOS-1718
+            self.task_state_start = self.get_master_task_state(dcos_api, self.tasks_start[0])
+            log.debug('Test app state at start:\n' + pprint.pformat(self.task_state_start))
```

Comment:
By `state comparison in the assertion`, I assume this is what you're talking about?
```
self.teamcity_msg.testFailed(
    "test_upgrade_vpc.test_task_state_remains_consistent",
    details="expected: {}\nactual:   {}".format(self.task_state_start, task_state_end))
```